File: Examples/CSharp/Data/AddOn/NamedRanges/CreateNamedRangeofCells.cs
using System;
using System.IO;

using Aspose.Cells;

namespace Aspose.Cells.Examples.CSharp.Data
{
    public class CreateNamedRangeofCells
    {
        //Output directory
        static string outputDir = RunExamples.Get_OutputDirectory();

        public static void Run()
        {
            // Opening the Excel file through the file stream
            Workbook workbook = new Workbook();

            // Accessing the first worksheet in the Excel file
            Worksheet worksheet = workbook.Worksheets[0];

            // Creating a named range
            Range range = worksheet.Cells.CreateRange("B4", "G14");

            // Setting the name of the named range
            range.Name = "TestRange";

            Style st = workbook.CreateStyle();
            st.Pattern = BackgroundType.Solid;
            st.ForegroundColor = System.Drawing.Color.Yellow;
            range.SetStyle(st);

            // Saving the modified Excel file
            workbook.Save(outputDir + "outputCreateNamedRangeofCells.xlsx");

            Console.WriteLine("CreateNamedRangeofCells executed successfully.");
        }
    }
}

File: Examples/VisualBasic/Data/AddOn/NamedRanges/CreateNamedRangeofCells.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells

Namespace Data.AddOn.NamedRanges
    Public Class CreateNamedRangeofCells
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Opening the Excel file through the file stream
            Dim workbook As New Workbook(dataDir & "book1.xls")

            ' Accessing the first worksheet in the Excel file
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Creating a named range
            Dim range As Range = worksheet.Cells.CreateRange("B4", "G14")

            ' Setting the name of the named range
            range.Name = "TestRange"

            ' Saving the modified Excel file
            workbook.Save(dataDir & "output.xls")
            ' ExEnd:1

        End Sub
    End Class
End Namespace